.github/workflows/validate.yml

```diff
@@ -21,8 +21,8 @@ jobs:
         uses: actions/setup-java@v2
         with:
           distribution: 'zulu'
-          java-version: 17
-          cache: 'gradle'
+          java-version: 20
```

Comment:
Since the target java version is properly set-up and the library will always be compatible down to java-8, it's safe to benefit from latest build improvements available in latest java version. 

This change is inspired by https://jakewharton.com/build-on-latest-java-test-through-lowest-java/

If the _policy_ is to use LTS version for some reason, I can easily revert that

Reply:
I guess that makes sense. We can keep this as is but revert compileVersion in the build.gradle so we don't force others to upgrade.

Reply:
Not having them in sync wouldn't be beneficial. The build is configured to be deterministic, so the version passed to `jvmToolchain` in `build.gradle` will be used to compile the app. If this specific version it's not installed, Gradle will attempt to download one. 
So if there was a _wrong_ version installed on the CI, the workflow would always have to download desired java version on each run, which would be suboptimal. Keeping CI config and build.gradle makes sure CI runs don't perform unnecessary actions.

Reply:
I kept Java 17 for now, but with the setup I added recently, it's safe to always use _latest_ java version and Gradle toolchains should make sure it's used in local builds, even if you don't have it installed manually. 

For example, my local setup has only Java 20 installed, and I compile the project with Java 17 downloaded by gradle:
<img width="473" alt="image" src="https://github.com/airbnb/lottie-android/assets/36954793/002dd8d1-0c50-4aad-ac19-06897d987320">
so there is no inconvenience in managing Java version + the build is fully deterministic 👌

Reply:
And to reply to https://github.com/airbnb/lottie-android/pull/2316#discussion_r1230182074 within a single thread. latest AGP version require Java 17, but this is just the minimum supported version
 
>you need to use JDK 17 or higher.

I think it's always encouraged to build with latest Java version to leverage latest build improvements (i.e. initially only Java 17 and up had native M1 chips support (and I believe only some of the distributions backported it to lower java versions eventually))